### PR TITLE
added endpoint to move agent from one agency to another

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

**Review by mingazhev — CHANGES_REQUESTED** [review]
Could not see counterparty change logic 
изображение

**Comment by mingazhev** [comment]
> 
> 
> Could not see counterparty change logic
> изображение

This is an error comment, sorry

File: Api/Models/Management/AuditEvents/AgentMovedFromOneAgencyToAnother.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        {
+            AgentId = agentId;
+            SourceAgencyId = sourceAgencyId;
+            DestinationAgencyId = destinationAgencyId;
+        }
+        
+        
+        public int AgentId { get; }
+        public int SourceAgencyId { get; }
+        public int DestinationAgencyId { get; }
```
Comment (mingazhev) [inline Api/Models/Management/AuditEvents/AgentMovedFromOneAgencyToAnother.cs:15]: `Target` looks more suitable as anthonym for `source`

File: Api/Services/Management/IChangeAgentAgencyService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+using System.Threading.Tasks;
+using CSharpFunctionalExtensions;
+
+namespace HappyTravel.Edo.Api.Services.Management
+{
+    public interface IChangeAgentAgencyService
```
Comment (mingazhev) [inline Api/Services/Management/IChangeAgentAgencyService.cs:6]: IAgentMovementService maybe?

File: Api/Services/Management/ChangeAgentAgencyService.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        public ChangeAgentAgencyService(EdoContext edoContext, IManagementAuditService managementAuditService)
+        {
+            _edoContext = edoContext;
+            _managementAuditService = managementAuditService;
+        }
+        
+        
+        public async Task<Result> Move(int agentId, int sourceAgencyId, int destinationAgencyId)
+        {
+            return await IsAgentExist()
```
Comment (mingazhev) [inline Api/Services/Management/ChangeAgentAgencyService.cs:22]: Method is named like returning bool, but returns Result instead

File: HappyTravel.Edo.Common/Enums/ManagementEventType.cs [hunk 4]
```diff
@@ -8,6 +8,7 @@ public enum ManagementEventType
         CounterpartyDeactivation = 30,
         AgencyDeactivation = 40,
         CounterpartyActivation = 50,
-        AgencyActivation = 60
+        AgencyActivation = 60,
+        MoveAgentFromOneAgencyToAnother = 70
```
Comment (mingazhev) [inline HappyTravel.Edo.Common/Enums/ManagementEventType.cs:12]: This should be a noun. `AgentMovement` maybe?

File: Api/Models/Management/AuditEvents/AgentMovedFromOneAgencyToAnother.cs (hunk 1 above)
Comment (DFSko) [reply Api/Models/Management/AuditEvents/AgentMovedFromOneAgencyToAnother.cs]: Fixed

File: Api/Services/Management/IChangeAgentAgencyService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Management/IChangeAgentAgencyService.cs]: Fixed

File: Api/Services/Management/ChangeAgentAgencyService.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Management/ChangeAgentAgencyService.cs]: Fixed

File: HappyTravel.Edo.Common/Enums/ManagementEventType.cs (hunk 4 above)
Comment (DFSko) [reply HappyTravel.Edo.Common/Enums/ManagementEventType.cs]: Fixed

File: Api/Controllers/AdministratorControllers/ManagementController.cs [hunk 5]
```diff
@@ -66,10 +67,29 @@ public async Task<IActionResult> RegisterAdministrator([FromBody] string invitat
...
+        
+        /// <summary>
+        /// Move agent from one agency to another
+        /// <param name="request">Change agent agency request</param>
+        /// </summary>
+        [HttpPost("change-agent-agency")]
+        [ProducesResponseType((int) HttpStatusCode.OK)]
+        [ProducesResponseType((int) HttpStatusCode.BadRequest)]
+        [AdministratorPermissions(AdministratorPermissions.AgentManagement)]
+        public async Task<IActionResult> MoveAgentToAgency([FromBody] ChangeAgentAgencyRequest request)
```
Comment (mingazhev) [inline Api/Controllers/AdministratorControllers/ManagementController.cs:80]: I think you should:
1. Move this to AgentsController
2. Change the route according to rest rules, with agent and agency in the path (as it is done in `SetSystemSettings` endpoint)
3. Return ProblemDetails for bad request

File: Api/Models/Management/ChangeAgentAgencyRequest.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        {
+            AgentId = agentId;
+            SourceAgencyId = sourceAgencyId;
+            DestinationAgencyId = destinationAgencyId;
+        }
+        
+        
+        public int AgentId { get; }
+        public int SourceAgencyId { get; }
+        public int DestinationAgencyId { get; }
```
Comment (mingazhev) [inline Api/Models/Management/ChangeAgentAgencyRequest.cs:15]: ...

File: Api/Services/Management/AgentMovementService.cs [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        public AgentMovementService(EdoContext edoContext, IManagementAuditService managementAuditService)
+        {
+            _edoContext = edoContext;
+            _managementAuditService = managementAuditService;
+        }
+        
+        
+        public async Task<Result> Move(int agentId, int sourceAgencyId, int targetAgencyId)
+        {
+            return await Result.Success()
```
Comment (mingazhev) [inline Api/Services/Management/AgentMovementService.cs:22]: Looks like you can remove this line

File: Api/Controllers/AdministratorControllers/ManagementController.cs (hunk 5 above)
Comment (DFSko) [reply Api/Controllers/AdministratorControllers/ManagementController.cs]: Fixed

File: Api/Models/Management/ChangeAgentAgencyRequest.cs (hunk 6 above)
Comment (DFSko) [reply Api/Models/Management/ChangeAgentAgencyRequest.cs]: Fixed

File: Api/Services/Management/AgentMovementService.cs (hunk 7 above)
Comment (DFSko) [reply Api/Services/Management/AgentMovementService.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Pull request merged**